### DISCUSSION TITLE
[2023-08-14] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by vitvakatu** [enso-org/enso#7580] [Weekly Demo Day]
As some team members are off today, I will summarize their progress for the last week.

@farmaazon:
- https://github.com/enso-org/enso/pull/7534 Component Browser does not render type names for instance members anymore. It greatly improves the readability of the list.
- https://github.com/enso-org/enso/pull/7497 Private items are hidden in the CB now.
- Joint effort with @somebody1234 to deliver new Top Bar with multiple useful buttons: https://github.com/enso-org/enso/pull/7488
![](https://user-images.githubusercontent.com/3919101/259426519-2d1fe4ad-6391-41d6-89a1-410492457b57.png)

[screen-recorder-tue-aug-08-2023-15-36-28.webm](https://github.com/enso-org/enso/assets/4046547/ac5ca74e-6a90-4fb1-acbc-e1b399d19de5)

@Frizi:
- https://github.com/enso-org/enso/pull/7507 will be finished later this week. In case of Rust panics we provide a nice message for users and allow them to report the issue.
- https://github.com/enso-org/enso/pull/7561 Positions of dropdowns are much more sensible now.
- Starting to work on https://github.com/enso-org/enso/issues/7434

![image](https://github.com/enso-org/enso/assets/6566674/e483febd-2f7f-4b1c-9486-65324686790d)

https://github.com/enso-org/enso/assets/919491/86678b9d-c04e-4545-8d11-2d1e85b9b951

Me:
- https://github.com/enso-org/enso/pull/7533 IDE now displays a correct user-provided user name. It will also solve the project name issue in the Cloud!
- https://github.com/enso-org/enso/pull/7550 We now have (limited) support for connections inside lambdas! 

https://github.com/enso-org/enso/assets/6566674/d4cb6733-dc9f-454e-ba1b-e76260bda467

https://github.com/enso-org/enso/assets/6566674/60af6413-e1b9-4e8c-a958-2906b5534d62

**Comment by mwu-tow** [comment]
Landed https://github.com/enso-org/enso/pull/7558, addressing issues with enso-project bundle opening.
Started investigating of Windows deep linking issue. https://github.com/enso-org/cloud-v2/issues/525

**Comment by MichaelMauderer** [comment]
*  Awaiting CI: #7508
![Peek 2023-08-09 16-07](https://github.com/enso-org/enso/assets/1428930/1a244e38-5c34-4c8b-8885-1cf84ac7b6a7)
*  Awaiting CI: #7532 
*  In Progress: #7312

**Comment by kazcw** [comment]
Implemented, awaiting QA: #7435

[vokoscreenNG-2023-08-09_22-29-13.webm](https://github.com/enso-org/enso/assets/1047859/2906599d-920f-44df-9c0f-c617ebbd5ecc)

In progress: handling context loss

**Comment by JaroslavTulach** [comment]
Performance improvements in `Vector` & `Array` - #7544. Example:

![Performance Improvements](https://github.com/enso-org/enso/assets/26887752/fbd4ddb1-0cf7-4640-b326-cc1df76e69c8)

Rename refactoring support is in waiting to be used - #7515
